extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -620,20 +623,35 @@ TemplatesAnalysisBuildItem analyzeTemplates(List<TemplatePathBuildItem> template
 
         // Register additional section factories
         if (engineConfigurations.isPresent()) {
+            // Use the deployment class loader - it can load application classes; it's non-persistent and isolated
+            ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+
             Collection<ClassInfo> sectionFactories = engineConfigurations.get().getConfigurations().stream()
                     .filter(c -> Types.isImplementorOf(c, Names.SECTION_HELPER_FACTORY, beanArchiveIndex.getIndex()))
                     .collect(Collectors.toList());
-            // Use the deployment class loader - it can load application classes; it's non-persistent and isolated
-            ClassLoader tccl = Thread.currentThread().getContextClassLoader();
-            for (ClassInfo factoryClass : sectionFactories) {
+            for (ClassInfo factoryInfo : sectionFactories) {
                 try {
-                    Class<?> sectionHelperFactoryClass = tccl.loadClass(factoryClass.toString());
+                    Class<?> sectionHelperFactoryClass = tccl.loadClass(factoryInfo.toString());
                     SectionHelperFactory<?> factory = (SectionHelperFactory<?>) sectionHelperFactoryClass
                             .getDeclaredConstructor().newInstance();
                     builder.addSectionHelper(factory);
-                    LOGGER.debugf("SectionHelperFactory registered during template analysis: " + factoryClass);
+                    LOGGER.debugf("SectionHelperFactory registered during template analysis: " + factoryInfo);
                 } catch (Exception e) {
-                    throw new IllegalStateException("Unable to instantiate SectionHelperFactory: " + factoryClass, e);
+                    throw new IllegalStateException("Unable to instantiate SectionHelperFactory: " + factoryInfo, e);
+                }
+            }
+
+            Collection<ClassInfo> parserHooks = engineConfigurations.get().getConfigurations().stream()
+                    .filter(c -> Types.isImplementorOf(c, Names.PARSER_HOOK, beanArchiveIndex.getIndex()))
+                    .collect(Collectors.toList());
+            for (ClassInfo parserHookInfo : parserHooks) {
+                try {
+                    Class<?> parserHookClass = tccl.loadClass(parserHookInfo.toString());
+                    ParserHook parserHook = (ParserHook) parserHookClass.getDeclaredConstructor().newInstance();
```

Comment:
Are abstract `ParserHook` classes out of this loop? Asking because `Types.isImplementorOf` seem to return true for abstract classes of a given interface

Reply:
Done.